### DISCUSSION TITLE
Too many steps to delete a repository

### DISCUSSION THREAD

**Discussion by Tatsh** [community/community#58337] [Repositories]
### Select Topic Area

Product Feedback

### Body

The deletion for repositories should be customisable. I should be able to turn off the excessive amount of steps. I do not like that I have to:

1. Click a confirmation button
2. Type the name of the repository (even selection of the text above the box is disabled)
3. Receive a second factor authentication if it has been a while

I would prefer that we can go back to just typing the name. And I do not think selection of the text above the box is necessary. It is honestly insulting as are the entirety of the steps above. This especially should not be happening for forks we are done with.

**Comment by the-officialjosh** [comment]
I share the same concerns and frustrations regarding the deletion process for repositories on GitHub. However, I understand that the current steps involved can be time-consuming and repetitive, especially when dealing with multiple repositories. 

Just Note the following however: 

The confirmation button serves as an additional security measure to prevent accidental deletions. Although it cannot be disabled by default, I acknowledge that it can feel redundant for users like yourself who frequently delete repositories. Similarly, typing the repository name is currently required to ensure that users explicitly specify the repository they want to delete. This step helps prevent unintended deletions, even though it may seem unnecessary for some users.

Regarding the second factor authentication (2FA), it is an essential security feature to protect your repositories from unauthorized access. However, I understand that it can be cumbersome during the deletion process. If you find it particularly inconvenient, you can consider adjusting your 2FA settings, although it's important to balance convenience with security.

**Comment by patrick-knight** [comment]
We hope deletion is an infrequent event and want make sure it was intended behavior. However, if speed is what you are after you could always use the [REST API to delete a repository](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#delete-a-repository)
